python/ray/data/iterator.py

```diff
@@ -676,13 +676,15 @@ def to_tf(
         feature_columns: Union[str, List[str]],
         label_columns: Union[str, List[str]],
         *,
+        additional_columns: Union[str, List[str]] = None,
         prefetch_batches: int = 1,
         batch_size: int = 1,
         drop_last: bool = False,
         local_shuffle_buffer_size: Optional[int] = None,
         local_shuffle_seed: Optional[int] = None,
         feature_type_spec: Union["tf.TypeSpec", Dict[str, "tf.TypeSpec"]] = None,
         label_type_spec: Union["tf.TypeSpec", Dict[str, "tf.TypeSpec"]] = None,
+        additional_type_spec: Union["tf.TypeSpec", Dict[str, "tf.TypeSpec"]] = None,
     ) -> "tf.data.Dataset":
         """Return a TF Dataset over this dataset.
 
```

Comment:
Hey @scottjlee, let me know if you think adding an example specifically for explaining the usage of the new argument is a good idea.

Reply:
yes good idea, could you add an example that demonstrates how to use `feature_type_spec`, `label_type_spec`, and `additional_type_spec`? thanks!

Reply:
Yeah definitely, will do!